src/lib.rs

```diff
@@ -182,61 +149,124 @@ impl<C: Config> Plugin for GgrsPlugin<C> {
+            .init_resource::<ConfirmedFrameCount>()
+            .init_resource::<MaxPredictionWindow>()
+            .init_resource::<RollbackOrdered>()
             .init_resource::<LocalPlayers>()
             .init_resource::<FixedTimestepData>()
             .add_schedule(GgrsSchedule, schedule)
             .add_schedule(ReadInputs, Schedule::new())
             .add_systems(PreUpdate, schedule_systems::run_ggrs_schedules::<C>)
-            .add_systems(LoadWorld, load_world)
-            .add_systems(SaveWorld, save_world);
+            .add_plugins((
+                GgrsSnapshotSetPlugin,
+                GgrsChecksumPlugin,
+                GgrsResourceSnapshotCopyPlugin::<Checksum>::default(),
+                GgrsEntitySnapshotPlugin,
+                GgrsComponentSnapshotReflectPlugin::<Parent>::default(),
+                GgrsComponentMapEntitiesPlugin::<Parent>::default(),
+                GgrsComponentSnapshotReflectPlugin::<Children>::default(),
+                GgrsComponentMapEntitiesPlugin::<Children>::default(),
+            ));
     }
 }
 
 /// Extension trait to add the GGRS plugin idiomatically to Bevy Apps
 pub trait GgrsApp {
-    /// Registers a component type for saving and loading from the world.
-    fn register_rollback_component<Type>(&mut self) -> &mut Self
+    /// Registers a component type for saving and loading from the world. This
+    /// uses [`Copy`] based snapshots for rollback.
+    fn rollback_component_with_copy<Type>(&mut self) -> &mut Self
     where
-        Type: GetTypeRegistration + Reflect + Default + Component;
+        Type: Component + Copy;
 
-    /// Registers a resource type for saving and loading from the world.
-    fn register_rollback_resource<Type>(&mut self) -> &mut Self
+    /// Registers a resource type for saving and loading from the world. This
+    /// uses [`Copy`] based snapshots for rollback.
+    fn rollback_resource_with_copy<Type>(&mut self) -> &mut Self
     where
-        Type: GetTypeRegistration + Reflect + Default + Resource;
+        Type: Resource + Copy;
+
+    /// Registers a component type for saving and loading from the world. This
+    /// uses [`Clone`] based snapshots for rollback.
+    fn rollback_component_with_clone<Type>(&mut self) -> &mut Self
+    where
+        Type: Component + Clone;
+
+    /// Registers a resource type for saving and loading from the world. This
+    /// uses [`Clone`] based snapshots for rollback.
+    fn rollback_resource_with_clone<Type>(&mut self) -> &mut Self
+    where
+        Type: Resource + Clone;
+
+    /// Registers a component type for saving and loading from the world. This
+    /// uses [`reflection`](`Reflect`) based snapshots for rollback.
+    ///
+    /// NOTE: Unlike previous versions of `bevy_ggrs`, this will no longer automatically
+    /// apply entity mapping through the [`MapEntities`](`bevy::ecs::entity::MapEntities`) trait.
+    /// If you require this behavior, see [`GgrsComponentMapEntitiesPlugin`].
```

Comment:
You hint towards the documentation of this plugin, but I cannot find any helpful text there that would further educate the user how to deal with this.

In addition to extra documentation, we probably need a parented example (e.g. cube with smaller child cube on top) that showcases usage of this.

Reply:
I agree, I've gone over all the plugins and added some extra documentation and at least 1 example demonstrating how to use the plugin. As a benefit, I've ensured every example properly compiles as a doc-test, so this has also increased test coverage.

As for the example, I completely agree. I think it would be good to maybe work on a stress-test example, as a benchmark. Since, as far as I can tell, there currently isn't really a benchmark in this crate for testing performance regression against.